src/client/minimap.cpp

```diff
@@ -683,8 +681,17 @@ MinimapMarker *Minimap::addMarker(scene::ISceneNode *parent_node)
 
 void Minimap::removeMarker(MinimapMarker **m)
 {
-	m_markers.remove_if([ptr = *m](const auto &up) { return up.get() == ptr; });
+	MinimapMarker *ptr = *m;
```

Comment:
What was wrong with remove_if? That it always has to scan the whole collection? (Whereas the loop can out early after the first match)

Reply:
`erase` on a vector will have to move all trailing elements in memory

Reply:
I'd prefer to keep it as is. There won't be terribly many markers anyway; the time complexity is linear either way.

The new code is more complex, and it is not clear why changing the order is acceptable. (Is it? Might it lead to (more) visual glitches when markers overlap?)